### ISSUE TITLE
CssSyntaxError at //-comment inside block

### ISSUE THREAD

**Issue by Niels-NTG** [stylelint/stylelint#5946]
### What steps are needed to reproduce the bug?

Any `.scss` file where there's a `// comment` inside a `{}` block gets flagged as "Unknown word (CssSyntaxError)".

An example file, see the comments:
```sass
// Does not cause an error

.columns {
    // Stylelint marks the first non-space character after the "//" with "Unknown word (CssSyntaxError)"
    display: flex;
    justify-content: space-between;
}

.column {
    @include user-select;
    $gutter-width: if($ismobile, 8px, 15px);

    // Comments of the same type do not get marked after Stylelint trips over the first one. Nor does it detects violations of rules such as "max-empty-lines" and "indentation" anymore if those are configured. This might indicate a bug that's deeper than just a big in the syntax validation itself.
    width: calc(50% - #{$gutter-width});

    &__sticky {
        position: sticky;
        top: 10px;
    }
}
```

This bug was not present in Stylelint 13.13.1, from which I just upgraded.

This issue is reminiscent of https://github.com/stylelint/stylelint/issues/3995 but would prefer not to work around bug by replacing all `//` comments with `/* */` since that's not what we decided on in our code style guide and both styles of comments are equally valid in SCSS/SASS.

### What Stylelint configuration is needed to reproduce the bug?

```json
{
    "plugins": [
        "stylelint-scss"
    ]
}
```

### How did you run Stylelint?

npx stylelint **/*.scss + the official Stylelint Visual Studio Code plugin

### Which version of Stylelint are you using?

14.5.1

### What did you expect to happen?

Comments in a SCSS file should be ignored by the linter unless they are not valid comments or break a comment-specific rule. In other words, if the linter  sees "// ", it should skip to the end of the comment.

### What actually happened?

The first non-space character after double slash // comments inside blocks get flagged as "Unknown word (CssSyntaxError)"

### Does the bug relate to non-standard syntax?

No
